public/components/common/query_utils/index.ts

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import dateMath from '@elastic/datemath';
+import { Moment } from 'moment-timezone';
+import { isEmpty } from 'lodash';
+import {
+  DATE_PICKER_FORMAT,
+  PPL_DEFAULT_PATTERN_REGEX_FILETER,
+} from '../../../../common/constants/explorer';
+import {
+  PPL_DATE_FORMAT,
+  PPL_INDEX_INSERT_POINT_REGEX,
+  PPL_INDEX_REGEX,
+  PPL_NEWLINE_REGEX,
+} from '../../../../common/constants/shared';
+
+/*
+ * "Query Utils" This file contains different reused functions in operational panels
+ *
+ * convertDateTime - Converts input datetime string to required format
+ */
+
+/**
+ * @param literal - string literal that will be put inside single quotes in PPL command
+ * @returns string with inner single quotes escaped
+ */
+const escapeQuotes = (literal: string) => {
+  return literal.replaceAll("'", "''");
+};
+
+export const convertDateTime = (
+  datetime: string,
+  isStart = true,
+  formatted = true,
+  isMetrics: boolean = false
+) => {
+  let returnTime: undefined | Moment;
+  if (isStart) {
+    returnTime = dateMath.parse(datetime);
+  } else {
+    returnTime = dateMath.parse(datetime, { roundUp: true });
+  }
+  if (isMetrics) {
+    const myDate = new Date(returnTime._d); // Your timezone!
+    const epochTime = myDate.getTime() / 1000.0;
+    return Math.round(epochTime);
+  }
+  if (formatted) return returnTime!.utc().format(PPL_DATE_FORMAT);
+  return returnTime;
+};
+
+const PROMQL_DEFAULT_AGGREGATION = 'avg';
+const PROMQL_CATALOG_ONLY_INDEX = /^(?<connection>[^|\s]+)\.(?<metric>\S+)$/i;
+const PROMQL_INDEX_REGEX = /(?<connection>[^|\s]+)\.query_range\('(?<promql>.+?)'/i;
+const PROMQL_AGG_ATTR_REGEX = /(?<aggregation>\S+) *?(by ?\((?<attributesGroupBy>.*?))\)? *?\((?<metric>\S+)\)/i;
```

Comment:
I know it's hard to see (well impossible to see).
Fixed a bug here.

Before/after regexes :
`/(?<aggregation>\S+) *?(by ?\((?<attributesGroupBy>.*?)\))? *?\((?<metric>\S+)\)/i`
`/(?<aggregation>\S+) *?(by ?\((?<attributesGroupBy>.*?))\)? *?\((?<metric>\S+)\)/i`